wordpress.org/public_html/wp-content/plugins/theme-directory/rest-api/class-theme-review-stats.php

```diff
@@ -0,0 +1,342 @@
+
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery
+		return $wpdb->get_results(
+			$wpdb->prepare(
+				"	
+				SELECT LEFT( upload_date, 7 ) as ym, AVG(days_to_review) as average_days_to_review
+
+				FROM (
+					SELECT p.post_name,
+					pm.meta_value,
+					substring_index( substring_index( pm.meta_value, '\"', 4 ), '\"', -1 ) AS upload_date,
+					p.post_date_gmt AS published_date,
+					DATEDIFF( p.post_date_gmt, substring_index( substring_index( pm.meta_value, '\"', 4 ), '\"', -1 ) ) as days_to_review
+			
+                    FROM {$wpdb->posts} p
+                    JOIN {$wpdb->postmeta} pm ON p.ID = pm.post_id AND pm.meta_key = '_upload_date'
+			
+					WHERE p.post_type = 'repopackage' and p.post_status = 'publish'
+			
+					ORDER BY p.ID desc
+				)a
+				
+				WHERE published_date > upload_date AND days_to_review IS NOT NULL AND upload_date >= %s
+
+				GROUP BY ym
+				ORDER BY upload_date ASC
+					",
+				$start_date
+			)
+		);
+	}
+
+	/**
+	 * Returns a list of published themes.
+	 *
+	 * @return void
+	 */
+	public function get_uploaded_themes() {
+		global $wpdb;
+
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery
+		return $wpdb->get_results(
+			"	
+			SELECT p.post_name as postName,
+				p.post_author as post_author,
+				substring_index( substring_index( pm.meta_value, '\"', 4 ), '\"', -1 ) AS author,
+				LEFT(p.post_date, 7) as published_on,
+				group_concat( t.slug SEPARATOR ', ' ) as tags
+
+			FROM {$wpdb->posts} p
+			JOIN {$wpdb->postmeta} pm ON p.ID = pm.post_id AND pm.meta_key = '_author'
+			LEFT JOIN {$wpdb->term_relationships} tr ON p.ID = tr.object_id
+			LEFT JOIN {$wpdb->term_taxonomy} tt ON tr.term_taxonomy_id = tt.term_taxonomy_id AND tt.taxonomy = 'post_tag'
+			LEFT JOIN {$wpdb->terms} t ON tt.term_id = t.term_id
+
+			WHERE p.post_type = 'repopackage' and p.post_status = 'publish'
+
+			Group by p.ID
+			Order By published_on asc
+			"
+		);
+	}
+
+	/**
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_REST_Response|WP_Error True if the request has read access for the item, WP_Error object otherwise.
+	 */
+	public function get_by_theme_type( $request ) {
+		$start_date = $this->get_start_date( $request );
+		$rows       = $this->get_uploaded_themes( $start_date );
+		$map        = array();
+
+		foreach ( $rows as $value ) {
+			$ym = $value->published_on;
+
+			if ( ! isset( $map[ $ym ] ) ) {
+				$map[ $ym ]['full-site-editing'] = 0;
+				$map[ $ym ]['classic']           = 0;
+			}
+
+			if ( str_contains( $value->tags, 'full-site-editing' ) ) {
+				$map[ $ym ]['full-site-editing']++;
+			} else {
+				$map[ $ym ]['classic']++;
+			}
+		}
+
+		$out = array();
+		foreach ( $map as $year_month => $value ) {
+			array_push( $out, array( $year_month, $value['full-site-editing'], $value['classic'] ) );
+		}
+
+		$out = $this->filter_items_before( $out, $start_date );
+
+		return new WP_REST_Response( array_values( $out ), \WP_Http::OK );
```

Comment:
You can just skip including the return code for the responses, I went looking for where a different return-code would be set... or just use `rest_ensure_response()` instead to handle it all magically.. 🤷 not important really this is fine.
```suggestion
		return rest_ensure_response( array_values( $out ) );
```